### PR TITLE
ASTT-106 add mypy pylint and coverage into the astt cam dependencies

### PR DESCRIPTION
**Pull request by USilere**
This MR is was to add mypy, pylint and coverage, only mypy and pylint were added.
Changes on the MR include:

- Added doctrings on classes and methods.
- Changed constant variables to uppercase as per PEP 8 style guide.
- Added mypy.ini file to configure mypy.
- Change max line on our lint configurations from 70 to 99.  
- Modularized functions on app.py (some to be moved to conftest and utils).
- Added UNRELEASED section on the CHANGELOG with all the changes on the branch.

To still be addressed:

- Adding pytest coverage
- Resolving disabled mypy and pylint issues

### CODE REVIEW

**Review by ssithole1 — COMMENTED** [review]
Work as expected on my local except I get this error when running the docker logs containerID

Screenshot 2024-09-02 at 10 19 22

**Comment by Silere8** [comment]
> Work as expected on my local except I get this error when running the docker logs containerID
> 
> Screenshot 2024-09-02 at 10 19 22

HI there Surpise I will check it out.

File: src/component_managers/start_simulator.py
Comment (AbednigoLethole) [inline src/component_managers/start_simulator.py:1]: It should be deleted. We no longer use it.

Merge from the main to ensure it's deleted.

File: tests/unit/test_modes.py [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+from src.component_managers.astt_comp_manager import ASTTComponentManager
 from src.component_managers.dish_modes import Mode
 
+# pylint: disable=invalid-name,unused-argument,too-many-public-methods
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=arguments-differ
+# pylint: disable=too-few-public-methods
+# pylint: disable=W0012
+# pylint: disable=duplicate-code
+
```
Comment (AbednigoLethole) [inline tests/unit/test_modes.py:15]: Should be addressed by decoupling the Component manager.

File: tests/unit/test_pointing.py [hunk 2]
```diff
@@ -1,32 +1,33 @@
 """Test the pointing function."""
 
+# pylint: disable=invalid-name,unused-argument,too-many-public-methods
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=broad-except
+
```
Comment (AbednigoLethole) [inline tests/unit/test_pointing.py:6]: It should not disabled but rather should be addressed.

File: tests/unit/test_sources.py [hunk 3]
```diff
@@ -1,5 +1,10 @@
+"""Test for sources in the sky."""
+
 # noqa: E501
-"""Test the sources functions."""
+# pylint: disable=invalid-name,unused-argument,too-many-public-methods
+# pylint: disable=attribute-defined-outside-init
+# mypy: disable_error_code="import-untyped"
+
```
Comment (AbednigoLethole) [inline tests/unit/test_sources.py:7]: It should be addressed, Not ignored.

File: src/component_managers/astt_comp_manager.py [hunk 4]
```diff
@@ -391,33 +346,25 @@ def track_sun(self, duration_time, az_speed=None, el_speed=None):
...
-            self.logger.info(
-                "---------------------------------------"
-            )
-            self.logger.info(
-                f"Sun_Az : {az_desired}, Sun_El : {el_desired}"
-            )
-            self.logger.info(
-                "---------------------------------------"
-            )
+            self.logger.info("---------------------------------------")
```
Comment (AbednigoLethole) [inline src/component_managers/astt_comp_manager.py:365]: Should be removed for the future

File: src/component_managers/astt_comp_manager.py [hunk 5]
```diff
@@ -391,33 +346,25 @@ def track_sun(self, duration_time, az_speed=None, el_speed=None):
...
+            self.logger.info(f"Sun_Az : {az_desired}, Sun_El : {el_desired}")
+            self.logger.info("---------------------------------------")
```
Comment (AbednigoLethole) [inline src/component_managers/astt_comp_manager.py:367]: It should be removed in the future

File: src/astt_gui/app.py [hunk 6]
```diff
@@ -1,5 +1,7 @@
-"""Facilitate integration between the GUI and system components."""
+# pylint: disable= W0603, W0718, W0602, E0401
+# mypy: disable_error_code="import-untyped"
```
Comment (AbednigoLethole) [inline src/astt_gui/app.py:2]: Should not be ignored but rather addressed, Maybe in the decoupling ticket.

File: src/component_managers/start_simulator.py
Comment (USilere) [reply src/component_managers/start_simulator.py:1]: Thanks, will remove it.

File: src/component_managers/astt_comp_manager.py (hunk 5 above)
Comment (USilere) [reply src/component_managers/astt_comp_manager.py:367]: We will have to work it on the decoupling ticket. I propose we setup a meeting with a Meerkat guy to understand how they give out info from an operation say slew, track or point.

File: src/astt_gui/app.py (hunk 6 above)
Comment (USilere) [reply src/astt_gui/app.py:2]: I agree. We should pick up the work with @ ssithole1 ASAP.

**Review by AbednigoLethole — APPROVED** [review]
Now, It looks good.

**Pull request merged**